src/BaseCurve.sol

```diff
@@ -13,16 +14,28 @@ import {UD60x18, ud60x18} from "@prb/math/UD60x18.sol";
  *      accomodating for the effect of fees, supply burn, airdrops, etc) are handled by the EthMultiVault instead
  *      of the curves themselves.
  */
-abstract contract BaseCurve {
+abstract contract BaseCurve is IBaseCurve {
+    /// @notice The name of the curve
     string public name;
 
+    /// @notice Construct the curve with a unique name
+    ///
+    /// @param _name Unique name for the curve
+    constructor(string memory _name) {
+        if (bytes(_name).length == 0) {
+            revert Errors.BaseCurve_EmptyStringNotAllowed();
+        }
+
+        name = _name;
+    }
+
     /// @notice The maximum number of shares that this curve can handle without overflowing.
     /// @dev Checked by the EthMultiVault before transacting
-    function maxShares() public view virtual returns (uint256);
```

Comment:
3.20 - Functions in Curve Contracts Could Be external Instead of public

Made this `external`